File: core/src/dc/targetman/epf/parts/AiPart.java
package dc.targetman.epf.parts;

import java.util.ArrayList;
import java.util.List;

import dc.targetman.ai.DefaultNode;
import dclib.util.Timer;

public final class AiPart {

	private static final float THINK_TIME = 0.1f;

	private final Timer updatePathTimer = new Timer(THINK_TIME, THINK_TIME);
	private List<DefaultNode> path = new ArrayList<DefaultNode>();

	public final List<DefaultNode> getPath() {
		return new ArrayList<DefaultNode>(path);
	}

	public final void setPath(final List<DefaultNode> path) {
		this.path = path;
	}

	public final boolean checkUpdatePath() {
		return updatePathTimer.check();
	}

	public final void tick(final float delta) {
		updatePathTimer.tick(delta);
	}

}

File: core/src/main/dc/targetman/epf/parts/AiPart.kt
package dc.targetman.epf.parts

import com.badlogic.gdx.math.MathUtils
import dc.targetman.ai.AiProfile
import dc.targetman.ai.Path
import dc.targetman.ai.SteerState
import dclib.util.Timer

class AiPart(val profile: AiProfile) {
    private val CALCULATE_PATH_TIME = MathUtils.random(0.07f, 0.12f)
    private val DETECT_TIME = MathUtils.random(0.01f, 0.015f)
    private val ALERT_TIME = 10f

    val path = Path()
    val steerState = SteerState()
    val waitTimer: Timer = Timer()
    val alertTimer = Timer(ALERT_TIME, ALERT_TIME)
    val sightTimer = Timer(Float.MAX_VALUE)

    private val calculatePathTimer = Timer(CALCULATE_PATH_TIME, CALCULATE_PATH_TIME)
    private val detectTimer = Timer(DETECT_TIME, DETECT_TIME)

    fun checkCalculatePath(): Boolean {
        return calculatePathTimer.check()
    }

    fun checkDetect(): Boolean {
        return detectTimer.check()
    }

    fun resetAlertTimer() {
        alertTimer.reset()
    }

    fun tick(delta: Float) {
        calculatePathTimer.tick(delta)
        detectTimer.tick(delta)
        waitTimer.tick(delta)
        alertTimer.tick(delta)
        sightTimer.tick(delta)
    }
}
